### ISSUE TITLE
TypeError: cannot pickle '_cffi_backend.FFI' object

### ISSUE THREAD

**Issue by scott2b** [tokusumi/fastapi-cloudauth#5]
Using a straight copy-paste of your example application, I get this pickle error.

```
Traceback (most recent call last):
  File "/Users/scott/.local/bin/uvicorn", line 10, in <module>
    sys.exit(main())
  File "/Users/scott/.local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/scott/.local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/scott/.local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/scott/.local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/scott/.local/lib/python3.8/site-packages/uvicorn/main.py", line 355, in main
    run(**kwargs)
  File "/Users/scott/.local/lib/python3.8/site-packages/uvicorn/main.py", line 379, in run
    server.run()
  File "/Users/scott/.local/lib/python3.8/site-packages/uvicorn/main.py", line 407, in run
    loop.run_until_complete(self.serve(sockets=sockets))
  File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
  File "/Users/scott/.local/lib/python3.8/site-packages/uvicorn/main.py", line 414, in serve
    config.load()
  File "/Users/scott/.local/lib/python3.8/site-packages/uvicorn/config.py", line 300, in load
    self.loaded_app = import_from_string(self.app)
  File "/Users/scott/.local/lib/python3.8/site-packages/uvicorn/importer.py", line 20, in import_from_string
    module = importlib.import_module(module_str)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "./main.py", line 12, in <module>
    @app.get("/", dependencies=[Depends(auth.scope("read:users"))])
  File "/Users/scott/.local/lib/python3.8/site-packages/fastapi_cloudauth/base.py", line 83, in scope
    obj = deepcopy(self)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 270, in _reconstruct
    state = deepcopy(state, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 270, in _reconstruct
    state = deepcopy(state, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 270, in _reconstruct
    state = deepcopy(state, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 270, in _reconstruct
    state = deepcopy(state, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/scott/.pyenv/versions/3.8.3/lib/python3.8/copy.py", line 161, in deepcopy
    rv = reductor(4)
TypeError: cannot pickle '_cffi_backend.FFI' object
```

**Comment by jurasofish** [comment]
I got this too and "fixed" it by recreating my condo Env. It only occurred after I installed some new packages which I didn't end up needing (I forget which). Happens when one of the classes returns a copy of itself. I might get around to a pr

**Comment by tokusumi** [comment]
How did you install this? Please tell me reproducible way.

**Comment by arrrrrmin** [comment]
Hello @tokusumi, yesterday I discovered this repo. Had the same problem. Also solved it via reinstalling the environment. Here are some requirements. (I'm moving my repo from Flask to fastAPI backend, just so you don't wonder ^^). I installed via `pip3 install fastapi-cloudauth`. I hope this helps you reproduce the error. Sadly I didnt freeze the requirements, so I can't tell you which versions I used. 

```
flask #removed after reinstallation
jinja2 #removed after reinstallation
gunicorn #removed after reinstallation
boto3 #removed after reinstallation
pandas #removed after reinstallation

fastapi
uvicorn
python-jose
requests
dotenv #removed after reinstallation
fastapi-cloudauth
```

**Comment by tokusumi** [comment]
I cannot reproduced 😢 .
I tried the followings commands in the configurations:

- Python: v3.8.3
- pip: v18.1
- application file: copy the example in readme and paste it in app.py

Notice that you must set valid region and user pool id in environment variables in the last line.

```sh
$ python3 -m venv env
$ source env/bin/activate
$ python -m pip install -U flask jinja2 gunicorn boto3 pandas
$ python -m pip install -U fastapi uvicorn python-jose requests python-dotenv
$ python -m pip install -U fastapi-cloudauth
$ export REGION=us-east-1 && export USERPOOLID=us-east-1_xxxxxxxxx && uvicorn app:app
INFO:     Started server process [20051]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
```

**Comment by arrrrrmin** [comment]
Ok, I reproduced it, in the above list I forgot about `authlib`, which was installed at the time:

```bash
$ python3 -m venv env
$ source env/bin/activate
$ python -m pip install --upgrade pip
$ python -m pip install flask jinja2 gunicorn boto3 pandas
$ python -m pip install fastapi uvicorn python-jose requests python-dotenv authlib
$ python -m pip install fastapi-cloudauth

**Comment by tokusumi** [comment]
Came across 👍 Thank you. I'll try to find a solution.

**Comment by tokusumi** [comment]
Hi, all!
I updated to resolve this problem in master and tested that in my reproducible environment. Could you check it with the latest?

**Comment by arrrrrmin** [comment]
@tokusumi `pip3 install git+https://github.com/tokusumi/fastapi-cloudauth.git` does't throw the pickle error anymore. Thanks for resolving :)

**Comment by tokusumi** [comment]
Thank you for helping us.
To fix this error, v0.1.3 was released.
